src/lib.rs

```diff
@@ -121,6 +134,25 @@ impl Builder {
         Ok(())
     }
 
+    /// Set to true to disable embedding a manifest 
+    pub fn set_no_embed(&self) -> Result<()> {
+        if let Ok(mut builder) = self.builder.try_write() {
+            builder.set_no_embed(true);
+        } else {
+            return Err(Error::RwLock);
```

Comment:
Question: Is this to poison the lock or... ?

Reply:
This is here because the uniffi tool we use to generate python bindings does not support mutable parameters. So we add an internal check to ensure there is only one mutable access at a time.

Reply:
We could use atomics here in `Buidler` to get around this eventually, if we end up having to have these binding layers hold on to a Mutex because of the Uniffi requirements of no mut vars as params. Not something that needs to be done for this PR, but it may make our lives easier down the road. 

```rust
use std::sync::atomic::{AtomicBool, Ordering};

struct Builder {
    no_embed: AtomicBool
}

impl Builder {
    pub fn set_no_embed(&self, value: bool) {
        self.no_embed.store(value, Ordering::Relaxed)
    }
}


fn main() {
    let b = Builder {
        no_embed: AtomicBool::new(true)
    };
    
    
    b.set_no_embed(true);
    
    // note how b, defined above, is not mutable. 
    std::thread::scope(|s| {
        let _ = s.spawn(|| b.set_no_embed(false));
        let _ = s.spawn(|| b.set_no_embed(true));
        let _ = s.spawn(|| b.set_no_embed(false));
    });
}
```